include/pybind11/attr.h

```diff
@@ -414,11 +414,15 @@ inline void check_kw_only_arg(const arg &a, function_record *r) {
         pybind11_fail("arg(): cannot specify an unnamed argument after a kw_only() annotation or args() argument");
 }
 
+inline void check_have_self_arg(function_record *r) {
```

Comment:
Maybe `append_self_arg_if_needed`?
"check" sounds like a const function that returns a bool.

Reply:
Updated.